sdk-docs/how-to/flask.md

```diff
@@ -0,0 +1,103 @@
+# Add an environment variable
+
+A charm configuration can be added if you require environment variables, for
+example, to pass a token for a service. For example, add the following
+configuration in `charmcraft.yaml`:
+
+```yaml
+config:
+  options:
+    token:
+      description: The token for the service.
+      type: string
+      required: true
+```
+
+PaaS App Charmer will map this as the `FLASK_TOKEN` environment variable. In
+general, the environment variable name is `FLASK_<config option name>` where the
+config option name will be converted to upper case. The configuration can be set
+on the deployed charm using `juju config <app name> token=<token>`
+
+# Database migration
+
+If your app depends on a database it is common to run a database migration
+script before app startup which, for example, creates or modifies tables. This
+can be done by including the `migrate.sh` script in the root of your project. It
+will be executed with the same environment variables and context as the flask
+app.
+
+If the migration script fails, the app won't be started and the app charm will
+go into blocked state. The migration script will be run on every unit and it is
+assumed that it is idempotent (can be run multiple times) and that it can be run
+on multiple units at the same time without causing issues. This can be achieved
+by, for example, locking any tables during the migration.
+
+# Including extra files in the OCI image
+
+The following files are included in the image by default from the root of the project:
+
+- `app`
+- `app.py`
+- `migrate`
+- `migrate.sh`
+- `migrate.py`
+- `static`
+- `templates`
+
+To change this, the following snippet needs to be added to the `rockfile.yaml`:
+
+```yaml
+parts:
+  flask-framework/install-app:
```

Comment:
Looking at the rockcraft.yaml I got from following the tutorial, there are a few more such parts. It feels like we should document all of them, really.

```yaml
parts:
  flask-framework/dependencies:
    plugin: python
    stage-packages:
    - python3-venv
    source: .
    python-packages:
    - gunicorn
    python-requirements:
    - requirements.txt
  flask-framework/install-app:
    plugin: dump
    source: .
    organize:
      app.py: flask/app/app.py
      templates: flask/app/templates
    stage:
    - flask/app/app.py
    - flask/app/templates
    prime:
    - flask/app/app.py
    - flask/app/templates
  flask-framework/config-files:
    plugin: dump
    source: /snap/rockcraft/1903/share/rockcraft/extensions/flask-framework
    organize:
      gunicorn.conf.py: flask/gunicorn.conf.py
  flask-framework/statsd-exporter:
    build-snaps:
    - go
    source-tag: v0.26.0
    plugin: go
    source: https://github.com/prometheus/statsd_exporter.git
  flask-framework/runtime:
    plugin: nil
    stage-packages:
    - ca-certificates_data
  pebble:
    plugin: nil
    stage-snaps:
    - pebble/latest/stable
    override-prime: |-
      craftctl default
      mkdir -p var/lib/pebble/default/layers
      chmod 777 var/lib/pebble/default
    stage:
    - bin/pebble
```

Reply:
yeh I agree we should do more documentation on this too

Reply:
I was thinking about this a bit more, the 2 pieces we have documented are relevant to users because they have to interact with it, the rest of what is there is probably not that relevant since users shouldn't really do anything with it if they want to keep using the extension

Reply:
Good to know, thanks!